server/core/tablecodec.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2016 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package core
+
+import (
+	"bytes"
+	"encoding/binary"
+
+	"github.com/juju/errors"
+)
+
+// TableIDDecoder defines method to extract tableID from key
+type TableIDDecoder interface {
+	DecodeTableID(key Key) int64
+}
+
+// DefaultTableIDDecoder is the default decoder.
+// unit test will use other mocked decoder.
+var DefaultTableIDDecoder defaultTableIDDecoder
+
+type defaultTableIDDecoder struct{}
+
+var tablePrefix = []byte{'t'}
```

Comment:
seem we should also consider table index. 

@disksing

Reply:
Just table prefix is ok. We have no plan to separate rows and indexes into different namespaces.